src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -8622,7 +8625,23 @@ internal static List<AlternateStreamData> GetStreams(string path)
             SafeFindHandle handle = NativeMethods.FindFirstStreamW(
                 path, NativeMethods.StreamInfoLevels.FindStreamInfoStandard,
                 findStreamData, 0);
-            if (handle.IsInvalid) throw new Win32Exception();
+            if (handle.IsInvalid)
+            {
+                // Except for a test for handle.IsInvalid, this must be immediately after a P/Invoke call.
+                int error = Marshal.GetLastWin32Error();
+                // Directories don't normally have alternate streams, so this is not an exceptional state.
+                // If a directory has no alternate data streams, FindFirstStreamW returns ERROR_HANDLE_EOF.
+                if (error == NativeMethods.ERROR_HANDLE_EOF && System.IO.Directory.Exists(path))
+                {
+                    return alternateStreams;
+                }
+                else
```

Comment:
We can remove the `else` because we have fast return in line 8636.

Reply:
Is this style okay here, or should I put it in a single line on line 8634 (the way most of the rest of the style issues you asked me to revert are)?

Reply:
The style's fine, I think; it matches the style we tend to work with in this repo. @iSazonov is just mentioning that we don't really need the `else` at all since we `return` in the `if`.

In general it's better to leave most style issues to separate PRs, but (at least in my opinion) anything you're adding/modifying directly in terms of functionality is fine to re-style to match what we're usually using in the repo, it just leaves less work to reformat later. I think Ilya's point was more that purely style changes should be left to a separate PR so it's clear what actual changes you're making in the code. 🙂

Reply:
I mean:
```c#
                if (error == NativeMethods.ERROR_HANDLE_EOF && System.IO.Directory.Exists(path))
                {
                    return alternateStreams;
                }

                // Filesystems other than NTFS generally don't support alternate streams, so we still
                // must be prepared to throw the error. It's usually ERROR_INVALID_PARAMETER in that case.
                throw new Win32Exception(error);
            }

Reply:
I was asking if I should have put `{ return alternateStreams; }` on the end of the line of the if, because this style (multiple lines for a single statement in a single block) doesn't match the remainder of the style that you asked me to revert.

```cs
                if (error == NativeMethods.ERROR_HANDLE_EOF) { return alternateStreams; }
                throw new Win32Exception(error);
```

Reply:
That said, I'm following vexx32's recommendation to make it match current style. New push (for this file) incoming. I haven't yet written the tests. (I know I should have written them first, but I'm just getting familiar with the code and I haven't even opened the test project to figure out how to write them.)

Reply:
We prefer to have new code right formatted. For the case:
```c#
                if (error == NativeMethods.ERROR_HANDLE_EOF)
                {
                  return alternateStreams;
                }

                throw new Win32Exception(error);
```

Reply:
Alright, new push has the correct formatting.